test/JDBC/input/ownership/BABEL-CROSS-DB-vu-prepare.mix

```diff
@@ -61,5 +67,112 @@ GO
 CREATE TABLE dbo.babel_cross_db_vu_prepare_db1_t3 (id int identity, a int);
 GO
 
+-- We have create a login and two user databases so far, now let's create
+-- users and schemas for that login in both the databases to test cross-db
+-- access. Also create cross-db views and functions containing as well.
+
+USE babel_cross_db_vu_prepare_db1;
+GO
+
+CREATE USER babel_cross_db_vu_prepare_u1 FOR LOGIN babel_cross_db_vu_prepare_l1;
+GO
+
+CREATE SCHEMA babel_cross_db_vu_prepare_s1 AUTHORIZATION babel_cross_db_vu_prepare_u1;
+GO
+
+USE babel_cross_db_vu_prepare_db2;
+GO
+
+CREATE USER babel_cross_db_vu_prepare_u2 FOR LOGIN babel_cross_db_vu_prepare_l1;
+GO
+
+CREATE SCHEMA babel_cross_db_vu_prepare_s2 AUTHORIZATION babel_cross_db_vu_prepare_u2;
+GO
+
+CREATE TABLE dbo.babel_cross_db_vu_prepare_t2 (id INT IDENTITY PRIMARY KEY, name VARCHAR(20));
+GO
+
+CREATE FUNCTION dbo.babel_cross_db_vu_prepare_f2 (@a int)
+    RETURNS INT AS
+    BEGIN
+        RETURN @a;
+    END;
+GO
+
+-- tsql user=babel_cross_db_vu_prepare_l1 password=12345678
+USE babel_cross_db_vu_prepare_db2;
+GO
+
+CREATE TABLE babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_t2 (id INT IDENTITY PRIMARY KEY, name VARCHAR(20));
+GO
+
+INSERT INTO babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_t2(name) VALUES('abc');
+GO
+
+CREATE FUNCTION babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_f2 (@a int)
+    RETURNS INT AS
+    BEGIN
+        RETURN @a;
+    END;
+GO
+
+CREATE VIEW babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_v2 AS
+    SELECT id, babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_f2(id), name
+    FROM babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_t2
+    ORDER BY id;
+GO
+
+CREATE PROCEDURE babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_p2
+AS
+    SELECT * FROM babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_t2 ORDER BY id;
+    SELECT * FROM babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_v2;
+GO
+
+USE babel_cross_db_vu_prepare_db1;
+GO
+
+CREATE TABLE babel_cross_db_vu_prepare_s1.babel_cross_db_vu_prepare_t1 (id INT IDENTITY PRIMARY KEY, name VARCHAR(20));
+GO
+
+INSERT INTO babel_cross_db_vu_prepare_s1.babel_cross_db_vu_prepare_t1(name) VALUES('def');
+GO
+
+-- View containing cross-db table and function on which login babel_cross_db_vu_prepare_l1 has privilege
+CREATE VIEW babel_cross_db_vu_prepare_s1.babel_cross_db_vu_prepare_v1 AS
+    SELECT x.id,
+        babel_cross_db_vu_prepare_db2.babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_f2(x.id),
+        x.name AS t1_name,
+        y.name AS t2_name
+    FROM babel_cross_db_vu_prepare_s1.babel_cross_db_vu_prepare_t1 x INNER JOIN
+    babel_cross_db_vu_prepare_db2.babel_cross_db_vu_prepare_s2.babel_cross_db_vu_prepare_t2 y
```

Comment:
can we add similar test where session user dont have permission on either of joined tables

Reply:
Done!